### PR TITLE
Merge wave-data-analysis into main branch with all MPA and ACA analyses running so far.

### PR DESCRIPTION
**Pull request by eco-ben**
All MPA and ACA analyses so far (without rugosity/turbidity in MPA currently) run successfully. 1_ steps take large amounts of time to run, and 2_ steps take around 15-20 minutes total.

### CODE REVIEW

File: src/1a_prep_MPA.jl [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+gbr_morphic_path = "$(MPA_DATA_DIR)/geomorphic/GBR10 GBRMP Geomorphic.tif"
+gbr_geomorphic = Raster(gbr_morphic_path, crs=EPSG(4326), lazy=true)
+
+# Load EPSG:4326 region features for subsetting of geomorphic and benthic rasters.
+regions_4326 = GDF.read(REGION_PATH_4326)
+
+# Reproject all raster files to GDA 2020
+# Avoids the need to repeatedly reproject within each step
+@showprogress dt = 10 "Prepping benthic/geomorphic/wave data..." for reg in REGIONS
```
Comment (ryulippmann) [inline src/1a_prep_MPA.jl]: Is comment line necessary?
```suggestion
# Reproject all raster files to GDA 2020
@showprogress dt = 10 "Prepping benthic/geomorphic/wave data..." for reg in REGIONS
```

File: src/1a_prep_MPA.jl [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    reg_idx_4326 = occursin.(reg[1:3], regions_4326.AREA_DESCR)
+
+    # Only create files that are needed
+    if !isfile(joinpath(MPA_OUTPUT_DIR, "$(reg)_bathy.tif"))
```
Comment (ryulippmann) [inline src/1a_prep_MPA.jl]: Should this comment go outside for loop?

File: src/1a_prep_MPA.jl [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+        write(joinpath(MPA_OUTPUT_DIR, "$(reg)_slope.tif"), target_slope; force=true)
+
+        target_slope = nothing
+        GC.gc()
+    end
+
+    if !isfile(joinpath(MPA_OUTPUT_DIR, "$(reg)_benthic.tif"))
+        # Load template raster data
+        bathy_gda2020_path = joinpath(MPA_OUTPUT_DIR, "$(reg)_bathy.tif")
```
Comment (ryulippmann) [inline src/1a_prep_MPA.jl]: Consistent comments with Rasters above/below

File: src/1b_prep_aca.jl [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+Ensure all rasters are the same size/shape for each region of interest.
+"""
+
+include("common.jl")
+
+# Loading regions_4326 for cropping of vector and raster data.
+regions_4326 = GDF.read(REGION_PATH_4326)
+
+# 1. Processing of geojson files into smaller-GBRMPA regions
+
```
Comment (ryulippmann) [inline src/1b_prep_aca.jl:15]: Consistent comment and formats to 1a

File: src/1b_prep_aca.jl [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+aca_bathy = Raster(aca_bathy_path, mappedcrs=EPSG(4326), lazy=true)
+
+aca_turbid_path = "$(ACA_DATA_DIR)/Turbidity-Q3-2023/turbidity-quarterly_0.tif"
+aca_turbid = Raster(aca_turbid_path, mappedcrs=EPSG(4326), lazy=true)
+
+@showprogress dt = 10 "Prepping bathymetry/turbidity/wave data..." for reg in REGIONS
+    reg_idx_4326 = occursin.(reg[1:3], regions_4326.AREA_DESCR)
+
+    if !isfile(joinpath(ACA_OUTPUT_DIR, "$(reg)_bathy.tif"))
+        # Cropping the raster speeds up the trim(mask()) step and reduces memory use - checked output in qgis
```
Comment (ryulippmann) [inline src/1b_prep_aca.jl]: Clean up commenting. To match 1a

File: src/2a_analyze_MPA.jl [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+            write(fpath, cache; force=true)
+        end
+
+        return nothing
+    end
+
+    function assess_region(reg)
+        # Load prepared bathymetry raster
+        src_bathy = Raster(joinpath(MPA_OUTPUT_DIR, "$(reg)_bathy.tif"); crs=EPSG(7844), lazy=true)
```
Comment (ryulippmann) [inline src/2a_analyze_MPA.jl]: Could this comment repetition be avoided and expressed once above all of these Raster reads

File: src/2b_analyze_ACA.jl [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+        suitable_slopes = nothing
+        res = nothing
+        result_raster = nothing
+        suitable_benthic = nothing
+        GC.gc()
+    end
+end
+
+@showprogress dt = 10 desc = "Analysing..." pmap(analyze_allen, REGIONS)
```
Comment (ryulippmann) [inline src/2b_analyze_ACA.jl]: American spelling consistency
```suggestion
@showprogress dt = 10 desc = "Analyzing..." pmap(analyze_allen, REGIONS)
```

File: src/3a_count_mpa.jl [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+    )
+    target_slopes = read(target_slopes .>= 95)
+
+    reefs = reef_features
+
+    for (target_row, reef) in enumerate(eachrow(reefs))
+        # Count number of locations that meet flats and slopes criteria.
+        # Have to loop over each reef individually as some geometries causes a crash.
+        # These should be safe to skip, and will resolve to `missing`.
+        if reef_features[target_row, :region] != ""
```
Comment (ryulippmann) [inline src/3a_count_mpa.jl]: Consider moving comment outside loop, and making more concise

File: src/3b_count_aca.jl [hunk 9]
```diff
@@ -3,45 +3,37 @@ Collate number of potentially suitable locations per reef, as defined by
 GBRMPA Features.
 """
```
Comment (ryulippmann) [inline src/3b_count_aca.jl]: Make docstrings dataset-specific for all files

File: src/3b_count_aca.jl [hunk 10]
```diff
@@ -57,26 +49,23 @@ function count_suitable(raster, reef)::Union{Int64, Missing}
...
 
 # Loop over each reef and count number of slopes and flats that meet criteria
-@showprogress dt=10 desc="Collating zonal stats..." for reg in REGIONS
-    # Load raster
+@showprogress dt = 10 desc = "Collating zonal stats..." for reg in REGIONS
+    # Load rasters and identify cells that are greater than or equal to 0.95
     target_flats = Raster(
-        joinpath(RESULT_DIR, "$(reg)_suitable_flats_85.tif"),
-        mappedcrs=EPSG(4326)
+        joinpath(ACA_OUTPUT_DIR, "$(reg)_suitable_flats.tif"),
```
Comment (ryulippmann) [inline src/3b_count_aca.jl]: Consitent with 3a. 
```suggestion
@showprogress dt = 10 desc = "Collating zonal stats..." for reg in REGIONS
    # Load rasters and identify cells that have at least 95% suitable cells in surrounding hectare for both flats and slopes
    target_flats = Raster(
        joinpath(ACA_OUTPUT_DIR, "$(reg)_suitable_flats.tif"),
```

File: src/3a_count_mpa.jl [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+@showprogress dt = 10 desc = "Collating zonal stats..." for reg in REGIONS
+    # Load rasters
+    target_flats = Raster(
+        joinpath(MPA_OUTPUT_DIR, "$(reg)_suitable_flats.tif"),
+        crs=EPSG(7844),
+        lazy=true
+    )
+    # Identify whether cells have 95% (or greater) of their surrounding hectare suitable
+    target_flats = read(target_flats .>= 95)
+
```
Comment (ryulippmann) [inline src/3a_count_mpa.jl]: Consistent with 3b
```suggestion
@showprogress dt = 10 desc = "Collating zonal stats..." for reg in REGIONS
    # Load rasters and identify cells that have at least 95% suitable cells in surrounding hectare for both flats and slopes
    target_flats = Raster(
        joinpath(MPA_OUTPUT_DIR, "$(reg)_suitable_flats.tif"),
        crs=EPSG(7844),
        lazy=true
    )
    target_flats = read(target_flats .>= 95)

```

File: src/3b_count_aca.jl [hunk 12]
```diff
@@ -57,26 +49,23 @@ function count_suitable(raster, reef)::Union{Int64, Missing}
...
-    # Any later transforms appear invalid (as it is no longer in EPSG:4326) and so leads
-    # to a crash.
-    # Taking a copy does not work, so a quick workaround is to simply read the geometries
-    # in again.
-    reefs = GDF.read(reef_path)
-    reefs.geometry = AG.reproject(reefs.geometry, GFT.EPSG(4326), crs(target_flats); order=:trad)
+    reefs = reef_features
 
     for (target_row, reef) in enumerate(eachrow(reefs))
         # Count number of locations that meet flats and slopes criteria.
```
Comment (ryulippmann) [inline src/3b_count_aca.jl]: As in 3b: Consider moving comment outside loop, and making more concise

File: src/3b_count_aca.jl [hunk 13]
```diff
@@ -107,32 +97,82 @@ end
...
+    # proportion of their area suitable for deployments
     target_reg = reef_features.region .== reg
     reef_features[target_reg, :flat_scr] .= round.(reef_features[target_reg, :flat_scr] / maximum(reef_features[target_reg, :flat_scr]), digits=4)
     reef_features[target_reg, :slope_scr] .= round.(reef_features[target_reg, :slope_scr] / maximum(reef_features[target_reg, :slope_scr]), digits=4)
 end
 
-# Have to write out results as shapefile because of ArcGIS not handling GeoPackages for
-# some reason...
+# Have to write out results as shapefile because of ArcGIS not handling GeoPackages
 GDF.write(
```
Comment (ryulippmann) [inline src/3b_count_aca.jl]:
```suggestion
# Write data to geopackage
GDF.write(
```

File: src/3b_count_aca.jl [hunk 14]
```diff
@@ -107,32 +97,82 @@ end
...
+    :flat_ha,
+    :slope_ha,
+    :Area_HA,
+    :n_flat,
+    :n_slope,
+    :flat_scr,
+    :slope_scr,
+    :UNIQUE_ID
+]]
+CSV.write(joinpath(ACA_QGIS_DIR, "potential_reef_areas.csv"), subdf)
```
Comment (ryulippmann) [inline src/3b_count_aca.jl:146]: Consistent with 3a: Comment and express subdf in consistent format.
```suggestion
# Write data to csv file
```
Either:
```suggestion
subdf = reef_features[:, [
    :region,
    :reef_name,
    :flat_ha,
    :slope_ha,
    :Area_HA,
    :n_flat,
    :n_slope,
    :flat_scr,
    :slope_scr,
    :UNIQUE_ID
]]
```
or
```suggestion
subdf = reef_features[:, [:region, :reef_name, :flat_ha, :slope_ha, :Area_HA, :n_flat, :n_slope, :flat_scr, :slope_scr, :UNIQUE_ID]]
```

**Comment by ryu-AIMS** [comment]
@BG-AIMS
Streamline/consistent comments across 1a & b.
I'd suggest opening together and checking comments in each to either match or highlight differences.

**Pull request merged**